projects/m2/015-the-sieve-of-eratosthenes/solutions/enricozoboli/solution_the_sieve_of_eratosthenes.py

```diff
@@ -0,0 +1,28 @@
+def list_prime(num):
+    """
+    This function take a limit number as parameter, lists all 
+    number from 0 to limit and return a list with only prime
+    numbers.
+    """
+    p = 2
+    number_list = []
+    prime_list = []
+    for i in range(0, num+1):
+        number_list.append(i)
+    while p < num:
+        # Starting at p*p improve the speed of the for loop because it doesn't
+        # need to restart from the first item of the list every time.
+        for x in range(p*p, num+1, p):
+            number_list[x] = 0
+        p += 1
+    for prime in number_list:
+        if 1 < prime:
+            prime_list.append(prime)
+        else:
+            continue
+    return prime_list
```

Comment:
I might be wrong but I don't see where the check of prime numbers is.

Reply:
Switching all multiple of p with zero and updating p in every iterations of the while isn't enough? Only because the check  slow down the program for high number(maybe there's another way but I didn't find it yet)